### PR TITLE
bell(oo) now returns oo

### LINKED ISSUE #9184: bell(n).limit(n, oo) should be oo rather than bell(oo)
`bell(n).limit(n,oo)` should take the value infinity, but the current output is `bell(oo)`. As the Bell numbers represent the number of partitions of a set, it seems natural that `bell(oo)` should be able to be evaluated rather than be returned unevaluated. This issue is also in line with the recent fixes to the corresponding limit for the Fibonacci numbers and Lucas numbers.

```
from sympy import *
n = symbols('n')
bell(n).limit(n,oo)

Output:
bell(oo)
```

I'm new to Sympy, so I'd appreciate the opportunity to fix this bug myself if that's alright.

### PR DESCRIPTION
**Pull request by sethebner**
fixed #9184 : namely evaluating bell(oo) as oo instead of leaving it unevaluated.

### CODE REVIEW

**Comment by moorepants** [comment]
Please edit the title to this PR to something more explanatory.

See these notes for guidelines: https://github.com/sympy/sympy/wiki/Development-workflow#writing-pull-request-title-and-description

**Comment by sethebner** [comment]
I have changed the pull request title. Should it be more explanatory, or is this fine?

**Comment by moorepants** [comment]
Yes, thanks. I don't know what `bell` is so someone else will have to comment on the actual code change.

File: sympy/functions/combinatorial/numbers.py [hunk 1]
```diff
@@ -424,6 +424,9 @@ def _bell_incomplete_poly(n, k, symbols):
 
     @classmethod
     def eval(cls, n, k_sym=None, symbols=None):
+        if n is S.Infinity:
```
Comment (zanzibar7) [inline sympy/functions/combinatorial/numbers.py]: `if n is S.Infinity and k_sym is None:`  will better handle the case of `bell(oo,x)`

File: sympy/functions/combinatorial/tests/test_comb_numbers.py [hunk 2]
```diff
@@ -103,6 +103,10 @@ def test_bell():
     m = Symbol('m', integer=True)
     assert bell(-1).evalf() == bell(m).rewrite(Sum).evalf(subs={m: -1})
 
+    # issue 9184
+    n = Dummy('n')
+    assert bell(n).limit(n, S.Infinity) == S.Infinity
```
Comment (zanzibar7) [inline sympy/functions/combinatorial/tests/test_comb_numbers.py]: I think `bell(oo) == S.Infinity` is simpler and sufficient.

**Comment by certik** [comment]
@sethebner would you mind addressing @zanzibar7's review? So that we can move forward and merge this PR.

File: sympy/functions/combinatorial/tests/test_comb_numbers.py (hunk 2 above)
Comment (certik) [inline sympy/functions/combinatorial/tests/test_comb_numbers.py]: Yes. Though the issue that this fixes (#9184) is using a limit, so it's nice to check that it actually works.

Comment (certik) [inline sympy/functions/combinatorial/tests/test_comb_numbers.py]: It should check both ways I think.

**Comment by gxyd** [comment]
@sethebner can you please address the requested changes.

**Pull request merged**